Clarify insider threats on "About SLSA" (#807)
slsa-framework/slsa: docs/spec/v1.0/about.md

In short: everyone involved in producing and consuming software, or providing
infrastructure for software.

**Software producers**, such as an open source project, a software vendor, or a
team writing first-party code for use within the same company. SLSA gives you
protection against tampering along the supply chain to your consumers, thereby

> reducing insider risk and increasing confidence that the software you produce

Reviewer: I see it as a fuzzy/fine line. It reduces risk and increase confidence that the software that you produce reaches your consumers as intended. It is "internal risk" whether by internal compromise or insider threat as I suspect the supply chain attack vectors are harder to exploit by someone who is not internal.

> reducing insider risk and increasing confidence that the software you produce

Reviewer 2: I think @mlieberman85 has a point. It does read as if this is specifically against insider risk. Should this read something like "reducing risk, including insider's, and increasing..."?

> reducing insider risk and increasing confidence that the software you produce

Author: Went with "both". Does that address your concern?

> reducing insider risk and increasing confidence that the software you produce

Reviewer 3: Yes! Sorry if it came off as pedantic. I think within context of the page it makes sense but I found that in the announcement it felt a bit unclear.

-   Code quality: SLSA does not tell you whether the developers writing the
    source code followed secure coding practices.
-   Producer trust: SLSA does not address organizations that intentionally
    produce malicious software, but it can reduce insider risks within an

>     organization you trust. SLSA's Build track protects against tampering during

Reviewer 4: ```suggestion
    organization you trust. SLSA's Build Track protects against tampering during
```

Reviewer 2: I like @olivekl's attention to these details!

-   Code quality: SLSA does not tell you whether the developers writing the
    source code followed secure coding practices.
-   Producer trust: SLSA does not address organizations that intentionally
    produce malicious software, but it can reduce insider risks within an

>     organization you trust. Currently SLSA's build track protects against

Reviewer: Nit: Production of malicious software isn't necessarily intentional.
```suggestion
-   Producer trust: While SLSA does not address trust that organizations
    won't produce malicious software, it can reduce insider risks within an
    organization you trust. Currently SLSA's build track protects against
```

Author: Sorry, I'm not sure I follow. What do you mean by unintentional malicious software? Is that not covered by "Code quality" above? My thinking is that it's valuable to call out both unintentional vulnerabilities (currently called "code quality") and intentional badness (currently called "producer trust").

Reviewer: I think that there are lots of nuances in the comment which is why I phrased it as a nit.

I would consider code quality to catch common mistakes like bugs and vulnerabilities in code more than malicious code. 

I think that the nit is that while _someone_ would ultimately be responsible for intentionally introducing malicious code, that individual/entity might not always be the producer/organization itself.

If the organization is a private company, then there could be potential risk of a compromised account being used to introduce malicious code (intentional from an adversary, unintentional from a company). If the organization is some open source entity, it may be possible for a "random" contributor to intentionally add malicious code that isn't identified (yes, I admit, this would likely fall at least partially within code quality and/or future directions).

I feel like we are trying to establish that SLSA cannot _create_ confidence in a producer/artifact in isolation. The SLSA specification is a tool that producers can use to help strengthen consumers' trust in an already trusted producer (as it supports the producer themselves to adhering to their best practices, guidelines, etc. by mitigating supply chain vulnerabilities).

Reviewer 2: What @arewm is saying sounds right to me and I'm supportive of his proposed change. I agree with him on the origin of the possible threat which is why I also responded to the point about "insider risk" @mlieberman85 raised. This said, this is merely the "about" section of the spec and although we should make sure it is not misrepresenting anything we have to accept that there is only so much we can address in a concise way. That's why I approved the PR as it stands anyway.

Author: Since there are enough approvals and I think everyone agrees that this is a strict improvement than before, I'm going to submit this as-is but not mark the underlying issue fixed. I do want to get agreement on this point because it gets at the heart of the scope of SLSA, so it's an important thing for us as an organization to agree upon.

I'll continue the discussion in the issue.

    tampering during or after the build; a
    [future SLSA Source track](future-directions#source-track) will also protect

>     against tampering before the build, particularly modifications to source
>     code.

Reviewer 5: Nit: it may be clearer if we say "unauthorised modification"?
```suggestion
    against tampering before the build, particularly unauthorized modification
    of source code.
```
Also, should we mention dependencies as a potential part of the source track? They seem to be top-of-mind as a risk for many.

Author: Will apply the suggestion. Isn't dependencies covered by the next bullet?

Reviewer 4: The next bullet is more about transitive dependencies, right? Not about risk to your immediate dependencies, but that a SLSA level doesn't take into account your dependency tree?

Author: Went with a variant of @olivekl's suggestion. PTAL

    source code followed secure coding practices.
-   Producer trust: SLSA does not address organizations that intentionally
    produce malicious software, but it can reduce insider risks within an
    organization you trust. SLSA's Build track protects against tampering during
    or after the build, and [future SLSA tracks](future-directions.md) intend to

>     protect against unauthorized modifications of source code and dependenies.

Reviewer 5: ```suggestion
    protect against unauthorized modifications of source code and dependencies.
```
